module/VuFind/src/VuFind/Db/Service/AbstractDbService.php

```diff
@@ -82,6 +85,11 @@ protected function getEntityClass(string $entity): string
      */
     public function persistEntity(EntityInterface $entity): void
     {
+        $config = $this->entityManager->getConfiguration();
+        $privacy = $config->Authentication->privacy ?? false;
+        if($privacy && $entity instanceof UserEntityInterface){
+            $this->getDbService(UserSessionPersistenceInterface::class)->addUserDataToSession($entity);
```

Comment:
You should return after persisting data to the session, because otherwise the function will proceed and save the data to the database, which is what we are trying to prevent.